src/commands/Misc/reddituser.ts

```diff
@@ -0,0 +1,219 @@
+					cutText(worstComment.body, 900)
+				].join('\n'));
+
+		return message.sendEmbed(embed);
+	}
+
+	private async fetchData(user: string, message: KlasaMessage) {
+		return Promise.all([this.fetchAbout(user, message), this.fetchComments(user, message), this.fetchSubmissions(user, message)]);
+	}
+
+	private async fetchAbout(user: string, message: KlasaMessage) {
+		const { data } = await fetch(`https://www.reddit.com/user/${user}/about/.json`, 'json')
+			.catch(() => { throw message.language.tget('COMMAND_REDDITUSER_QUERY_FAILED'); }) as Reddit.Response<'about'>;
+		return data;
+	}
+
+	private async fetchComments(user: string, message: KlasaMessage, after = '', dataElements: Reddit.CommentDataElement[] = []) {
+		const url = new URL(`https://www.reddit.com/user/${user}/comments.json`);
+		url.searchParams.append('after', after);
+		url.searchParams.append('limit', '100');
+
+		const { data } = await fetch(url, 'json')
+			.catch(() => { throw message.language.tget('COMMAND_REDDITUSER_QUERY_FAILED'); }) as Reddit.Response<'comments'>;
+
+		for (const child of data.children) {
+			dataElements.push(child.data);
+		}
+
+		if (data.children.length === 100) await this.fetchComments(user, message, data.children[99].data.name, dataElements); // Recurse to get more data
+
+		return dataElements;
+	}
+
+	private async fetchSubmissions(user: string, message: KlasaMessage, after = '', dataElements: Reddit.PostDataElement[] = []) {
+		const url = new URL(`https://www.reddit.com/user/${user}/submitted.json`);
+		url.searchParams.append('after', after);
+		url.searchParams.append('limit', '100');
+
+		const { data } = await fetch(url, 'json')
+			.catch(() => { throw message.language.tget('COMMAND_REDDITUSER_QUERY_FAILED'); }) as Reddit.Response<'posts'>;
+
+		for (const child of data.children) {
+			dataElements.push(child.data);
+		}
+
+		if (data.children.length === 100) await this.fetchSubmissions(user, message, data.children[99].data.name, dataElements); // Recurse to get more data
+
+		return dataElements;
+	}
+
+	private calculateControversiality(comments: Reddit.CommentDataElement[]) {
+		if (!comments.length || comments.length < 5) return 0;
+		let count = 0;
+
+		for (const comment of comments) {
+			if (comment.controversiality === 1) count++;
+		}
+
+		return (count / comments.length) * 100;
+	}
+
+	private calculateKincaid(sentences: number, words: number, syllables: number) {
+		const sentenceWeight = 0.39;
+		const wordWeight = 11.8;
+		const adjustment = 15.59;
+
+		return (
+			(sentenceWeight * (words / sentences)) +
+			(wordWeight * (syllables / words)) -
+			adjustment
+		);
+	}
+
+	private calculateSyllables(word: string) {
+		word = word.toLowerCase();
+		if (word.length <= 3) return 1;
+		word = word.replace(/(?:[^laeiouy]es|ed|[^laeiouy]e)$/, '');
+		word = word.replace(/^y/, '');
+		const syl = word.match(/[aeiouy]{1,2}/g);
+
+		return syl ? syl.length : 1;
```

Comment:
`String.prototype.replace` returns the modified string, chain it:
```typescript
const syl = word.replace(/(?:[^laeiouy]es|ed|[^laeiouy]e)$/, '')
	.replace(/^y/, '')
	.match(/[aeiouy]{1,2}/g);
```

Reply:
And while this is complex, please put a note above to clarify what the algorithm actually does and some examples of why this works.